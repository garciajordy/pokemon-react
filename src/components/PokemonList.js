import React, { useState, useEffect } from 'react';
import axios from 'axios';
import Pokemon from './Pokemon';
import Pagination from './Pagination';

const apiUrl = 'https://pokeapi.co/api/v2/pokemon/';
function PokemonList() {
  const [pokemon, setPokemon] = useState([]);
  const [currentPageUrl, setCurrentPageUrl] = useState(apiUrl);
  const [nextPageUrl, setNextPageUrl] = useState();
  const [previousPageUrl, setPreviousPageUrl] = useState();
  const [loading, setLoading] = useState(true);

  useEffect(() => {
    setLoading(true);
    axios.get(currentPageUrl)
      .then((res) => {
        setLoading(false);
        setNextPageUrl(res.data.next);
        setPreviousPageUrl(res.data.previous);
        setPokemon(res.data.results.map((p) => p));
      });
  }, [currentPageUrl]);

  function goToNextPage() {
    setCurrentPageUrl(nextPageUrl);
  }

  function goToPrevPage() {
    setCurrentPageUrl(previousPageUrl);
  }

  if (loading) return 'Loading...';
  return (
    <div>
      <div className="pokeContainer">
        {pokemon.map((poke) => (
          <Pokemon key={poke.name} name={poke.name} url={poke.url} />
        ))}
      </div>
      <Pagination
        goToNextPage={nextPageUrl ? goToNextPage : null}
        goToPrevPage={previousPageUrl ? goToPrevPage : null}
      />
    </div>
  );
}

export default PokemonList;
