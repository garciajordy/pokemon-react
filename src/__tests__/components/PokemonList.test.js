import React from 'react';
import { render, cleanup, screen } from '@testing-library/react';
import { Provider } from 'react-redux';
import PokemonList from '../../components/PokemonList';
import store from '../../store/index';

afterEach(cleanup);

it('should take a snapshot', () => {
  const { asFragment } = render(<Provider store={store}><PokemonList /></Provider>);
  expect(asFragment(<Provider store={store}><PokemonList /></Provider>)).toMatchSnapshot();
});

it('displays heading', () => {
  render(<Provider store={store}><PokemonList /></Provider>);
  expect(screen.getByText('Write the exact name')).toBeTruthy();
});
