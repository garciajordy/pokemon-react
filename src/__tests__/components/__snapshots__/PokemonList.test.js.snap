// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should take a snapshot 1`] = `
<DocumentFragment>
  <div>
    <div
      class="search-box"
    >
      <input
        placeholder="Search..."
        tpye="text"
      />
       Write the exact name
    </div>
    <div
      class="pokeContainer"
    />
    <div
      class="button-bottom"
    >
      First PageLast Page
    </div>
  </div>
</DocumentFragment>
`;
